### DISCUSSION TITLE
HOGWARTS: Deleted Cards Printing Undefined due to nested arrays when pushed to a new array

### DISCUSSION THREAD

**Discussion by robertncole1** [nss-evening-cohort-14/evening-client-side#72] [Help Me!] [answered]
Your Name: Rob Cole

Have you searched the other help tickets? Yes

Describe your issue: I am able to assign my cards a unique ID and also delete them and print the new card to the expelled students div, but it's printing undefined and I am not sure why.

List any error messages that appear: No errors appear

What have you googled?
I have tried to figure out how to add items to the new array and found different answers: 
https://stackoverflow.com/questions/4156101/copy-array-items-into-another-array - concat method?
I tried putting my splice method inside a push to new array, which seems to be working but not printing

My JS code:
https://github.com/robertncole1/sorting-hat/blob/student-delete/main.js

**Comment by drteresavasquez** [comment]
@robertncole1 let's jump into the Bear Cave to have a look.

**Comment by robertncole1** [comment, marked as answer]
My issue was that I needed to put a spread operator (...) on my .push method. So instead of targeting an id, I spliced the array and then added it to a new array of expelled students. But when I did that, it copied an entire array into the new array when I just wanted an object. To get around that, we used the spread syntax which can be used when all elements from an object or array need to be included in a list of some kind. 

So for example:
```
const deleteStudent = (e) => {
  const targetType = e.target.type;
  const targetId = e.target.id;
  if (targetType === "button") {
    let expelled = students.splice(targetId, 1);
    expelledStudents.push(...expelled); //spread
  }
  studentBuilder(students);
  expelledBuilder(expelledStudents);
  console.log(expelledStudents)
};
```

**Reply by drteresavasquez** [reply]
NOICE!!!! 🥇

**Reply by chiestroud** [reply]
Thank you for asking this question Rob! I had the same issue and I actually did something different to remove unidentified by adding [0] like ${item[0].name} and ${item[0].house}. This method works like magic too but I was not able to explain it (Console.log returns Array [0] so I added [0]. That was all I could say). Anyway, after listening through your explanation and Dr. T's comments, it clicked! Thank you so much :)
